baybe/searchspace/continuous.py

```diff
@@ -42,6 +44,36 @@ class SubspaceContinuous(SerialMixin):
     )
     """List of linear inequality constraints."""
 
+    def __repr__(self) -> str:
+        """Override the standard __repr__."""
+        if self.is_empty:
+            return ""
+        # Convert the lists to dataFrames to be able to use pretty_printing
+        par_df = parameter_cartesian_prod_to_df(self.parameters)
```

Comment:
Hey @AdrianSosic, The reason why I used the cartesian function is that some attributes, like 'parameters', were defined as lists. To be able to use `pretty_printing` I needed to convert them into `dataFrames` and using the cartesian function was the easiest way because the function was already defined.
But we do not have to leave it this way, I am open for better suggestions :)

Reply:
Yes, but you see, there are a number of conceptual problems with your approach:
* First of all, you are abusing the `parameter_cartesian_prod_to_df` method. You can see already by the name that it is intended for parameters only, and if you look at the signature, you'll accordingly see that it expects an `Iterable[Parameter]`. However, you also pass it constraints and similar stuff. Right now, we don't have `mypy` enabled for the search spaces, but once it's active, it would also complain.
* If it really was necessary to use that function, then the "right" approach would be to generalize the function such that it is no longer coupled to paramaters only. Then, you could use it at least from a type safety perspective.
* However, if your printing is set up correctly, I doubt that any such computations would be necessary. You see, it's just conceptually weird that your printing requires to "compute" stuff instead of using stuff that is already there (how can it be that for printing purposes computations are necessary if the object itself does NOT require these computations?). What I mean: logically, printing should only require access to already existing stuff. If stuff does not already exist, then it's 1) either not relevant for the object and hence not relevant for printing or 2) it's an indication that it should be added to the object (which I doubt) and then again no computations would be needed in the printing.

Reply:
Yes, but you see, there are a number of conceptual problems with your approach:

* First of all, you are abusing the parameter_cartesian_prod_to_df method. You can see already by the name that it is intended for parameters only, and if you look at the signature, you'll accordingly see that it expects an Iterable[Parameter]. However, you also pass it constraints and similar stuff. Right now, we don't have mypy enabled for the search spaces, but once it's active, it would also complain.
* If it really was necessary to use that function, then the "right" approach would be to generalize the function such that it is no longer coupled to paramaters only. Then, you could use it at least from a type safety perspective.
* However, if your printing is set up correctly, I doubt that any such computations would be necessary. You see, it's just conceptually weird that your printing requires to "compute" stuff instead of using stuff that is already there (how can it be that for printing purposes computations are necessary if the object itself does NOT require these computations?). What I mean: logically, printing should only require access to already existing stuff. If stuff does not already exist, then it's 1) either not relevant for the object and hence not relevant for printing or 2) it's an indication that it should be added to the object (which I doubt) and then again no computations would be needed in the printing.

Reply:
this is absolutely right, this function cannot be used here, due i) its only for parameters and not constraints and ii) its exessively expensive and iii) the cartesian product doesnt make sense for conti parameters hence also not for SubspaceContinuous

it also cannot be used for cosntraints, it will probably fail when tried with a non-empty list of constraints

What you effectively print here for `parameters` is the discrete substpace, which is the cartesian product of all parameter values. instead I would suggest a table summarizing the parameter objects, like e.g.

**1) Parameter Summary**
```
Name | Num. Values | Type | Encoding
MyParam1 | 10 | NumericalDiscrete... | -
MyParam2 | 3 | Categorical... | OHE
```
etc (pretty formatted) for `SubspaceDiscrete` 
and
```
Name | Lower Bound | Upper Bound
MyParam3 | -50 | 50
MyParam4 | 0 | 100
```
for `SubspaceContinuous`

**2) Constraints Summary**
Secondly, this recipe could also be repeated for Constraints
```
Type | Affected Parameters
SumContraint | ['MyParam1', 'MyParam2']
CustomContraint | ['MyParam1']
```

**3) Excerpt of the Searchspace (only for SubspaceDiscrete)**
Thirdly, then you could print an excerpt of the actual searchspace via pretty printing `self.exp_rep`. This only makes sense for `SubspaceDiscrete`

**4) Other minor fields summary**

**5) Custom Additions**
e.g.
- how many point of the discrete space have already been recommended / measured, basically a metadata summary
- whatever else comes to mind we can flexibly add

Reply:
This is something we should probably discuss in a meeting, but I already want to bring this here: Several of the changes that you propose here @Scienfitz would require us to already include some variants of pretty printing for `Parameters` and `Constraints`, which was not the intended goal of this PR. I however fully agree that we *should* also have pretty printing for these in general.
My suggestion: Just use something that is better than what we have now and works without having too much trouble with the params (e.g. just print their name and class), and change this portion in an upcoming PR where we include pretty printing for Parameters, allowing us to basically just use their `__str__` method. Same for constraints.